x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/bucket_nesting_editor.test.tsx

```diff
@@ -0,0 +1,211 @@
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: false }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={jest.fn()}
+      />
+    );
+
+    expect(component.children().length).toBe(0);
+  });
+
+  it('should display a dropdown with the parent column selected if 3+ buckets', () => {
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['c', 'a', 'b'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0, operationType: 'count', isBucketed: true }),
+            b: mockCol({ suggestedPriority: 1, operationType: 'max', isBucketed: true }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: true }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={jest.fn()}
+      />
+    );
+
+    const control = component.find('[data-test-subj="indexPattern-nesting-select"]').first();
+
+    expect(control.prop('value')).toEqual('c');
+  });
+
+  it('should reorder the columns when a column is selected in the dropdown', () => {
+    const setColumns = jest.fn();
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['c', 'a', 'b'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0, operationType: 'count', isBucketed: true }),
+            b: mockCol({ suggestedPriority: 1, operationType: 'max', isBucketed: true }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: true }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={setColumns}
+      />
+    );
+
+    const control = component.find('[data-test-subj="indexPattern-nesting-select"]').first();
+    (control.prop('onChange') as (e: unknown) => {})({
+      target: { value: 'b' },
+    });
+
+    expect(setColumns).toHaveBeenCalledWith(['c', 'b', 'a']);
+  });
+
+  it('should move to root if the first dropdown item is selected', () => {
+    const setColumns = jest.fn();
+    const component = mount(
+      <BucketNestingEditor
+        columnId="a"
+        layer={{
+          columnOrder: ['c', 'a', 'b'],
+          columns: {
+            a: mockCol({ suggestedPriority: 0, operationType: 'count', isBucketed: true }),
+            b: mockCol({ suggestedPriority: 1, operationType: 'max', isBucketed: true }),
+            c: mockCol({ suggestedPriority: 2, operationType: 'min', isBucketed: true }),
+          },
+          indexPatternId: 'foo',
+        }}
+        setColumns={setColumns}
+      />
+    );
+
+    const control = component.find('[data-test-subj="indexPattern-nesting-select"]').first();
+    (control.prop('onChange') as (e: unknown) => {})({
+      target: { value: '' },
+    });
+
+    expect(setColumns).toHaveBeenCalledWith(['a', 'c', 'b']);
+  });
```

Comment:
Some test ideas that I don't see, but you can add if you would like:

- Test that toggling the switch twice ends in the same state as before
- Test that toggling the switch enables the switch in the other dimension
- Test the case where it's all buckets- should be able to replace the last column